http3: flaky TestClientResponseValidation

```
=== RUN   TestClientResponseValidation/invalid_headers
    client_test.go:201: 
        	Error Trace:	D:/a/quic-go/quic-go/http3/http3_helper_test.go:141
        	            				D:/a/quic-go/quic-go/http3/client_test.go:201
        	            				D:/a/quic-go/quic-go/http3/client_test.go:175
        	Error:      	An error is expected but got nil.
        	Test:       	TestClientResponseValidation/invalid_headers
--- FAIL: TestClientResponseValidation (0.01s)
    --- PASS: TestClientResponseValidation/HEADERS_frame_too_large (0.01s)
    --- FAIL: TestClientResponseValidation/invalid_headers (0.01s)

--- a/http3/client_test.go
+++ b/http3/client_test.go
@@ -198,6 +198,9 @@ func testClientResponseValidation(t *testing.T, tr *Transport, rsp []byte, expec
 	select {
 	case err := <-errChan:
 		expectStreamWriteReset(t, str, expectedReset)
+		// The client closes the stream after sending the request,
+		// so we need to wait for the RESET_STREAM frame to be received.
+		time.Sleep(scaleDuration(10 * time.Millisecond))
 		expectStreamReadReset(t, str, expectedReset)
 		return err
 	case <-time.After(time.Second):

